.github/workflows/build.yml

```diff
@@ -0,0 +1,26 @@
+name: ble.sh build
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  builds:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - name: Package
+        run: |
+          mkdir ble.sh
+          make install INSDIR="$(pwd)/ble.sh"
+          tar -cJf ble.tar.xz ble.sh
+      - name: Release
+        uses: softprops/action-gh-release@v1
+        with:
+          files: ble.tar.xz
```

Comment:
Hmm, can we put a dynamically generated name here (i.e., including the current commit hash as in the above suggestion)? If not, I'm fine with `ble-nightly.tar.xz`.

Reply:
I think it's possible with setting environment variables, I will try

Reply:
I can solve this with a glob, but it will create different tarballs with different hashes

Reply:
> I can solve this with a glob,

Ah, OK. That's smart.

> but it will create different tarballs with different hashes

Yeah, that was one of the things that I wanted to test locally. Maybe this is a bad idea, but I wanted to keep all the past distributed versions. If they will be accumulated on the release page of `nightly`, my first concern is how these tarballs are ordered on the release page. Also, the latest tarball needs to be easily found by the users at a glance. In addition, the latest version should be at the top or the bottom. Maybe we can instead use the date such as `$(date +%Y%m%d)` (though, which still overwrites the previous version)?

Reply:
It is a bit unconventional to keep past distributed versions of a nightly release, since it is a moving target. May I ask why do you want to keep the past distributed versions? Maybe I can think of another way instead of keeping all the versions in the nightly release.

Reply:
> It is a bit unconventional to keep past distributed versions of a nightly release,

I'm not sure if this is unconventional on the GitHub workflows, but outside the GitHub, I have been thinking that it's common to keep all the recent nightly builds (though we may drop the very old nightly builds).

> May I ask why do you want to keep the past distributed versions?

There are no strong reasons, but I have a feeling that we should keep the distributed versions, in case there are any problems on the user side, in order to better identify the problem and to keep the reproducibility.

> Maybe I can think of another way instead of keeping all the versions in the nightly release.

Thanks. I'll also try it later by creating an experimental repository on GitHub.

Reply:
> I'm not sure if this is unconventional on the GitHub workflows, but outside the GitHub, I have been thinking that it's common to keep all the recent nightly builds (though we may drop the very old nightly builds).

I think because git already retains all the versions, so we don't need to do it ourselves.

> There are no strong reasons, but I have a feeling that we should keep the distributed versions, in case there are any problems on the user side, in order to better identify the problem and to keep the reproducibility.

We can have a version file that contains the hash in the tarball, it will make it easy to retrieve the version that a user is using.

Reply:
I'm caring about the problems arising in the build stage in a specific environment, which are not included in the git history.

Reply:
Currently, only 1 environment is used, which is Ubuntu-20.04, and later when you use more environments and some builds fail, you can always check the log of that build. The log also includes the hash the of commit that the action is using.

Reply:
Hmm, sorry. I'm still talking about the problems on the user side as mentioned in https://github.com/akinomyoga/ble.sh/pull/197#discussion_r895239307. I didn't mean the build failure but the possibility of the problems silently embedded in the build stage without any error message. More specifically, when problems appear on the side of the users who downloaded a specific version of nightly builds, we want to isolate the probability that the build result of the nightly build has some problems. I think the probability is not so large but not zero, so we want to keep the way to confirm that it is not the cause.

Reply:
> We can have a version file that contains the hash in the tarball, it will make it easy to retrieve the version that a user is using.
Then is this a viable solution?

Reply:
> Then is this a viable solution?

Maybe I miss your point, but how does this solve it? My point is that, even if it is based on the same commit, the build results might be different depending on the setup of the building environment.

Reply:
I have pushed experimental support for the archive name to `uyha:master` (04aa800). This is experimental, i.e., we may drop it later if we finally concluded we wouldn't need it.

Reply:
I confirmed that when multiple files are added, these files are ordered in alphabetical order.

Reply:
I still think that it is not necessary to include the hash in the package since it doesn't provide more information compared to the output of `ble version`, and it makes it more difficult for scripting purpose (say a script that always downloads latest version has to make calls to GitHub API to retrieve the information and try to find the newest version)

Reply:
> I still think that it is not necessary to include the hash in the package since it doesn't provide more information compared to the output of `ble version`

The purpose of including the hash in the package name is not to provide the information to the user, but to keep the data of tarball for ourselves for the later support. As far as we use the same name, the old build result seems to be overwritten in the release page. For example, when the user have reported the information obtained from `ble version`, how can we recover the tarball that the user has downloaded?

> it makes it more difficult for scripting purpose (say a script that always downloads latest version has to make calls to GitHub API to retrieve the information and try to find the newest version)

OK, this is a good point. I haven't so far intended to make it possible, but it becomes possible if we provide the build result with a fixed URL. How about uploading the tarball with two names, i.e., one with the hash including the hash to retain the data and the other without the hash for convenience of the user.

Reply:
> I was thinking that we can use the hash to get the commit, then use the build steps defined in the workflow to reproduce the tarball of that build.

I think it should work basically, but we cannot make it 100% sure that the building environment will never be changed or the build result will not be different due to some accidental reasons (server timeouts, possible race conditions, etc.). If it is possible, it is easier for me to keep the actually distributed tarball.

f848d8c is experimental support. Can you see the result https://github.com/uyha/ble.sh/releases/tag/nightly?

Reply:
Yes, I think that should be good

Reply:
Thanks. I'm going rebase and squash some commits.